compiler/src/main/java/com/squareup/anvil/compiler/codegen/ContributesSubcomponentHandlerGenerator.kt

```diff
@@ -245,21 +246,33 @@ internal class ContributesSubcomponentHandlerGenerator(
   }
 
   private fun generateFactory(
-    factoryDescriptor: ClassDescriptor
+    factoryDescriptor: ClassDescriptor,
+    contribution: Contribution,
+    module: ModuleDescriptor
   ): TypeSpec {
+    val superclass = factoryDescriptor.asClassName()
+
     val builder = if (DescriptorUtils.isInterface(factoryDescriptor)) {
       TypeSpec
         .interfaceBuilder(SUBCOMPONENT_FACTORY)
-        .addSuperinterface(factoryDescriptor.asClassName())
+        .addSuperinterface(superclass)
     } else {
       TypeSpec
         .classBuilder(SUBCOMPONENT_FACTORY)
         .addModifiers(ABSTRACT)
-        .superclass(factoryDescriptor.asClassName())
+        .superclass(superclass)
     }
 
     return builder
       .addAnnotation(Subcomponent.Factory::class)
+      .addAnnotation(
+        // This will allow injecting the factory instance.
+        AnnotationSpec
+          .builder(ContributesBinding::class)
+          .addMember("scope = %T::class", contribution.parentScope.asClassName(module))
+          .addMember("boundType = %T::class", superclass)
+          .build()
```

Comment:
I know it's the equivalent of the Obama medal meme, but I love how Anvil annotations can be reused within generated code to avoid even more boilerplate and it just works.

Reply:
Oh thats really interesting, didn't even know you could do that!

Reply:
Actually, I should give the medal to @RBusarow, because he fixed an issue for `@ContributesBinding`.

Reply:
But then as the author of the frameworks I used to fix it, I think you can claim at least some of that medal as well. 😀